modules/sdk/gradle-plugins-change-log-builder/README.markdown

```diff
@@ -0,0 +1,90 @@
+# Change Log Builder Gradle Plugin
+
+The Change Log Builder Gradle plugin allows you to generate and maintain a
+change log file based on the Git commits on your project. A change log file
+generated by this plugin looks like this:
+
+```ini
+#
+# Bundle Version 1.0.1
+#
+9c77ff4c95cb1a325db3bdd089be105206e8b63c^..b421f00ac84b065685b131833fecc594fc01c760=LPS-123 LPS-1321
+
+#
+# Bundle Version 1.0.2
+#
+b421f00ac84b065685b131833fecc594fc01c760^..bc15d8d84e12b9544f78e4e3743c510dbaec2d89=LPS-456
+```
+
+Every time the [`buildChangeLog`](#buildchangelog) task is executed, a new line
+is added to the change log, which lists all Git [commit prefixes](#ticketidprefixes)
+(usually issue ticket IDs) that occurred in a certain range. The end of the
+range is always the tip of the current branch, while the start is:
+
+- the first time `buildChangeLog` is executed: the most recent commit from at least two years ago;
+- otherwise: the range end of the last line in the change log.
```

Comment:
Every line of the changelog is `commit range = list of tickets occurred in that range`. The first time we run `buildChangeLog`, the change log file does not exist yet, but we still want to list all the tickets that occurred to the project until the current moment in time. So we ask Git to give us the hash of the most recent commit that happened 2 years ago, and use that as range start.

The `buildChangeLog` task is executed manually right before publishing a new version of the module. We want to add a new line in the change log with all the tickets that occurred between the last release and the one we're going to do. So we read the change log file, get the range end of the last line (which was the `HEAD` at the moment of the last release) and use that as range start for the current one.